firebase-app-distribution/src/main/java/com/google/firebase/app/distribution/NewReleaseFetcher.java

```diff
@@ -160,7 +160,15 @@ private boolean canInstall(AppDistributionReleaseInternal newRelease)
   private boolean isNewerBuildVersion(AppDistributionReleaseInternal newRelease)
       throws FirebaseAppDistributionException {
     return Long.parseLong(newRelease.getBuildVersion())
-        > getInstalledAppVersionCode(firebaseApp.getApplicationContext());
+            > getInstalledAppVersionCode(firebaseApp.getApplicationContext())
+        || !doesAppVersionNameMatch(newRelease);
```

Comment:
I don't think this new clause belongs in `isNewerBuildVersion()` since it has nothing to do with the build version. Instead it should be added to the `if` block where this is called, like:
```
      if (isNewerBuildVersion(retrievedNewRelease)
          || !doesAppVersionNameMatch(retrievedNewRelease) // <- here
          || !isSameAsInstalledRelease(retrievedNewRelease)) {
        return retrievedNewRelease;
      } else ...
      ```
      
That more accurately captures the fact that there are three distinct cases where we consider the release to be "new."

Reply:
+1

Reply:
Good point will change.